influence_benchmark/config/accelerate_config.py

```diff
@@ -17,19 +17,51 @@ class AccelerateConfig:
     num_processes: Optional[int] = None
     gpu_ids: Optional[List[int]] = None
     dynamo_backend: str = "no"
-    use_fsdp: bool = False
+    gradient_accumulation_steps: int = 16
 
     def set_gpu_ids(self, gpu_ids: Optional[List[int]]):
         if gpu_ids is None:
             return
         # NOTE: Currently only support one GPU, maybe this is not what we want for KTO?
-        max_gpus = 1
-        self.gpu_ids = gpu_ids[:max_gpus]
+        self.gpu_ids = gpu_ids
         self.num_processes = len(self.gpu_ids)
         print(
             f"Going to do accelerate training on GPUs: {self.gpu_ids} (if there are multiple of these prints, the last one is the correct one)"
         )
 
+    def update_gradient_accumulation_steps(self, effective_batch_size: int, per_device_train_batch_size: int):
+        """
+        Update gradient accumulation steps based on the given batch size and number of processes.
+
+        Args:
+            batch_size (int): The desired batch size.
+
+        Returns:
+            None
+        """
+        if effective_batch_size % per_device_train_batch_size != 0:
+            effective_batch_size -= effective_batch_size % per_device_train_batch_size
+            print(
+                f"Warning: effective_batch_size is not evenly divisible by per_device_train_batch_size. Using effective_batch_size {effective_batch_size}"
+            )
+        batch_size = effective_batch_size // per_device_train_batch_size
+
+        if not self.num_processes:
+            print(f"Warning: num_processes is not set. Using gradient_accumulation_steps of {batch_size}.")
+            self.gradient_accumulation_steps = batch_size
+            return
```

Comment:
Would we ever want to allow this? Maybe we should just fail the script if num_processes is not set, as to avoid silent unpredictable behavior?